go-controller/pkg/ovn/base_network_controller_pods.go

```diff
@@ -279,6 +282,15 @@ func (bnc *BaseNetworkController) findPodWithIPAddresses(needleIPs []net.IP) (*k
 		if util.PodCompleted(p) || util.PodWantsHostNetwork(p) || !util.PodScheduled(p) {
 			continue
 		}
+
+		// If the network type is Layer 3, then IP allocation is per node, so we can filter pods by node name.
+		// If pod is not applicable to live migration, restrict the search to only pods on the filtered node
+		// Note we could check to see if pod has been migrated, but that search is even heavier than the
+		// alternative.
+		if bnc.TopologyType() == ovntypes.Layer3Topology && !kubevirt.IsPodLiveMigratable(p) && len(nodeName) > 0 && nodeName != p.Spec.NodeName {
```

Comment:
For the default network the topology is an empty string but you still want it covered in this check right?

Reply:
wait why do we set topology type as empty for default network...shouldnt we set it to L3?

Reply:
I guess because it was conceptually thought as a different topology given that it has all the extra things that the secondary network doesn't have. And because the NetConf of the default network doesn't really have the attribute `TopologyType` so the natural default value for that is the the zero value for string.

This is an example of the typical check we have been using for this same thing
```
func (na *NodeAllocator) hasNodeSubnetAllocation() bool {
	// we only allocate subnets for L3 secondary network or default network
	return na.netInfo.TopologyType() == types.Layer3Topology || !na.netInfo.IsSecondary()
}
```

Reply:
ok as we spoke about on slack, I'd prefer to make minimal changes at this point (yeah I'm paranoid). I added a comment with a TODO to fix this later.